core/state_transition.go

```diff
@@ -873,14 +873,12 @@ func (st *stateTransition) innerExecute() (*ExecutionResult, error) {
 	}
 	effectiveTipU256, _ := uint256.FromBig(effectiveTip)
 
-	shouldCheckGasFormula := true
 	if st.evm.Config.NoBaseFee && msg.GasFeeCap.Sign() == 0 && msg.GasTipCap.Sign() == 0 {
 		// Skip fee payment when NoBaseFee is set and the fee fields
 		// are 0. This avoids a negative effectiveTip being applied to
 		// the coinbase when simulating calls.
-		shouldCheckGasFormula = false
 	} else {
-
+		shouldCheckGasFormula := !st.msg.SkipNonceChecks && !st.msg.SkipFromEOACheck
```

Comment:
In what situation should we skip the check?  It looks like it is called in estimate/eth_call methods?

Reply:
yeah, estimate/eth_call will set both to `true`, and trace_call/access_list.